### PR TITLE
CXP-822 Mapping scope to acls

### PR DESCRIPTION
**Pull request by RiffVentura**
**Description (for Contributor and Core Developer)**

Added a scope to acls mapper

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by tseho — APPROVED** [review]
Regarding the `const` vs `registry`, the proposed solution is fine for me, it's a perfectly good first step.
If needed, we will iterate over the implem.

File: src/Akeneo/Tool/Bundle/ApiBundle/Security/ScopeToAclMapper.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    ];
+
+    /**
+     * @return string[]
+     */
+    public function getAllScopes(): array
+    {
+        return array_keys(self::ScopeAclMap);
+    }
+
```
Comment (tomglvng) [inline src/Akeneo/Tool/Bundle/ApiBundle/Security/ScopeToAclMapper.php:97]: phpdoc return

File: src/Akeneo/Tool/Bundle/ApiBundle/spec/Security/ScopeToAclMapperSpec.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            'write_channel_settings',
+            'read_association_types',
+            'write_association_types',
+            'read_products',
+            'write_products',
+            'delete_products',
+        ]);
+    }
+
+    public function it_throw_on_unknown_scope()
```
Comment (tomglvng) [inline src/Akeneo/Tool/Bundle/ApiBundle/spec/Security/ScopeToAclMapperSpec.php:40]:
```suggestion
    public function it_throws_on_unknown_scope()
```

**Review by fitn — APPROVED** [review]

**Review by tomglvng — APPROVED** [review]

**Comment by tseho** [comment]
I'm merging without the CI.
The current PR only add a new class (which is not used yet) and the corresponding unit test.
It's the third time in a row that the legacy behat are failing on unrelated things.
https://app.circleci.com/pipelines/github/akeneo/pim-community-dev/25586/workflows/02c84736-65d0-494f-8b7b-6e4c764de20f/jobs/113914

**Pull request merged**